code/processes/dqe.q

```diff
@@ -1,34 +1,36 @@
 \d .dqe
 
-/- default parameters
-subscribeto:@[value;`subscribeto;`];                          /-list of tables subscribed to
-subscribesyms:@[value;`subscribesyms;`];                      /-list of syms subscribed to
-schema:@[value;`schema;0b];                                   /-retrieve schema from tickerplant
-tickerplanttypes:@[value;`tickerplanttypes;`tickerplant];     /-list of tickerplant types to try and make a connection to
-replaylog:@[value;`replaylog;0b];                             /-replay the tickerplant log file
-tpconnsleepintv:@[value;`tpconnsleepintv;10];                 /-default wait time before trying to reconnect to discovery/TP
-/- end of default parameters
-
-subscribe:{[]
-  if[count s:.sub.getsubscriptionhandles[tickerplanttypes;();()!()];
-  .lg.o[`subscribe;"found available tickerplant, attempting to subscribe"];
-  /-set the date that was returned by the subscription code i.e. the date for the tickerplant log file
-  /-and a list of the tables that the process is now subscribing for
-  subinfo: .sub.subscribe[subscribeto;subscribesyms;schema;replaylog;first s];
-  /-setting subtables and tplogdate globals
-  .dqe,:subinfo]
-  }
-
 init:{
-  .servers.CONNECTIONS:distinct .servers.CONNECTIONS, .dqe.tickerplanttypes;
   .lg.o[`init;"searching for servers"];
-  .servers.startup[];
-  .servers.startupdependent[tickerplanttypes;tpconnsleepintv];
-  subscribe[];
+  .servers.startup[];                                                                                           /- Open connection to discovery
   }
 
-tableExists:{x in tables[]};                                  /-function to check for table, param is table name as a symbol
+gethandles:{exec procname,proctype,w from .servers.SERVERS where (procname in x) | (proctype in x)};
+
+tableexists:{x in tables[]};                                                                                    /- function to check for table, param is table name as a symbol
+
+runcheck:{[fn;vars;rs]                                                                                          /- function used to send other function to test processes
+  fncheck:(` vs fn);
+  if [0b=b:fncheck[2] in key value .Q.dd[`;fncheck[1]];                                                         /- run check to make sure passed in function exists
+    .lg.e[`function;"Function ",(string fn)," doesn't exist"];
+    :()];
+
+  rs:(),rs;                                                                                                     /- set rs to a list
+  h:.dqe.gethandles[rs];                                                                                        /- check if processes exist and are valid
+  
+  missingproc:rs where not rs in h[`procname],h[`proctype];                                                     /- check all process exist
+  failedproc:{x,`failed}'[missingproc];
```

Comment:
This doesn't seem to do much.

Reply:
This stores the processes from the input that are not up or exist.
Right now it is only used for the error message but will be used in one of the next parts.

Reply:
> This stores the processes from the input that are not up or exist

aren't those already stored in `missingproc` though?

Reply:
Get your point, this just adds failed to each for an upsert later.
I will remove it and replace it with a more useful expression later :)